pkg/virt-operator/util/canary.go

```diff
@@ -0,0 +1,62 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package util
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/cache"
+
+	v1 "kubevirt.io/client-go/apis/core/v1"
+	"kubevirt.io/kubevirt/pkg/controller"
+)
+
+const (
+	canaryDaemonSetAnnotation = "kubevirt.io/canary-daemonset"
+)
+
+func IsCanaryDaemonSet(daemonSet *appsv1.DaemonSet) bool {
+	_, exists := daemonSet.Annotations[canaryDaemonSetAnnotation]
+	return exists
+}
+
+func GetCanaryPod(kv *v1.KubeVirt, daemonSet *appsv1.DaemonSet, infraPods cache.Store) *k8sv1.Pod {
+	for _, obj := range infraPods.List() {
+		pod := obj.(*k8sv1.Pod)
+		owner := controller.GetControllerOf(pod)
+		_, isCanary := pod.Annotations[canaryDaemonSetAnnotation]
```

Comment:
I don't think you need this annotation, You're just looking for a pod in a "ready" state that's using the updated image.